src/libs/Navigation/linkTo.ts

```diff
@@ -153,14 +153,14 @@ export default function linkTo(navigation: NavigationContainerRef<RootStackParam
         const isTargetNavigatorOnTop = topRouteName === action.payload.name;
 
         const isTargetScreenDifferentThanCurrent = Boolean(topmostCentralPaneRoute && topmostCentralPaneRoute.name !== action.payload.params?.screen);
-        const areParamsDifferent = !shallowCompare(topmostCentralPaneRoute?.params, action.payload.params?.params);
```

Comment:
@nkdengineer I don't think we should remove `shallowCompare` here because it also handles use cases of search pages. You can check this intermediate commit https://github.com/Expensify/App/commit/a72d2217e7a0bf7f4a9b9e6515a18fce49f60cd5 for information. 

Can we fix this issue by improving the `shallowCompare` method? Like just skipping parameter with value `undefined`

Reply:
@eh2077 i updated please check again